proposal/WarClans612/proposal.rst

```diff
@@ -0,0 +1,171 @@
+
+The main idea is to use past text and tag pairs to learn the intrinsic information and 
+perform tag generation for future text content. If possible, the newly generated text will
+then be supervised by human to ensure its validity for further model training.
+
+BackPropagation is the backbone for how the Deep Learning Approach could *learn* from 
+existing information. In fitting the neural network, BackPropagation will compute gradient of
+loss function with respect to the weights of the network for every input output pairs.
+
+By numerically optimizing the Neural Network (can be think as complex mathematical functions),
+the model could learn to minimize error towards the real expected output.
+
+Algorithm to be Used
+-----------------
+Extreme Deep Learning Classification refers to the problem of assigning to each document
+its most relevant subset of class labels from an extremely large label collection, where
+the number of labels could reach hundreds of thousands or millions. Multi-label classification
+is fundamentally different from the traditional binary or multi-class classification problems
+which have been intensively studied in the machine learning literature.
+
+Due to huge size of Deep Learning Approach, this method is usually hard to be implemented
+for mobile device, therefore, it will enable the user to choose if the implemented model 
+will be used for low powered device. The change will be small, which is changing
+Convolutional Layer into Depthwise Separable Convolution with the trade of lesser computation
+need with a bit lower accuracy for the results.
+
+The paper, MobileNets: Efficient Convolutional Neural Networks for Mobile Vision Applications,
+discuss about changing the convolutional layer into Deptwise Separable Convolutional for better
+performance in mobile device.
+
+Perspective users
+=================
+
+- Content Provider to generate tag for each content for easier recommendation
+- Social Media to generate tag to know the topic for each post (possibly prevent malicious usage or even for related content recommendation)
+- Forum Admin to generate tag for each content in its responsibility
+- Medical Worker to generate tag (e.g. possible diseases) from patient written record
+
+System architecture
+===================
+
+Input (when initializing):
+
+- Data containing past input and generated tag
+- All possible list of generated tag
+- If the model will be run on mobile device, we could improvise to shrink the model
+
+Output (when initialization):
+
+- Model that is ready to be used for generating tag
+
+Input (for usage after initialization):
+
+- Text content that needed tag generation
+
+Output (for usage after initialization):
+
+- List of possible tags
+
+Modularization
+-----------------
+
+- User API (only this layer is supposed to be accessed by user)
+- Normal model definition Module
+- Mobile enabled model definition Module
+- Training Module
+- Validation Module
+- Model saving module
```

Comment:
Do you only save, or is this about both saving and loading?  If it's for the bidirectional I/O, what is the difference between the next item 'data management module'?

It will be good if you elaborate these questions or make adjustment based on your answer (in your mind).

Reply:
It should be for saving and loading the model (which is trained using the initial data).
It is quite different with Data Management Module which is responsible for Loading the data and converting it into the format which the model need to learn according to the hyperparameter.

Reply:
Thanks.  It is clear.

You could consider to add the elaboration to the proposal, if you want to be reminded later.  It's also ok to just leave the proposal as is.